webpack/assets/javascripts/react_app/components/componentRegistry.js

```diff
@@ -72,11 +72,20 @@ const componentRegistry = {
     return Object.keys(this.registry).join(', ');
   },
 
-  defaultWrapper(component, data = null, store = null, flattenData = false) {
+  defaultWrapper(
+    component,
+    data = null,
+    metadata = null,
+    store = null,
+    flattenData = false,
+    context = null
+  ) {
     const factory = this.wrapperFactory();
 
     factory.with('i18n');
-
+    if (context) {
+      factory.with('context', metadata);
```

Comment:
Just to make sure I understand, 
would the component have access to the global context which was declared in ReactApp?
Does this override its data? Thanks

Reply:
there are two ways to mount a component:
1. using `mount_react_component` via erb
2. using react-router under `ReactApp` root
this factory creates a mounter wrapped by the context, it's the same context that used in `reactapp` root, both point to the same context object so the data is the same

Reply:
doesn't adding this here also wrap all of the components imported from the registry, even if they are already under the `ReactApp` root?

Reply:
No, this is wrapped only for components that mounted via `mount_react_component` through erb files. so components that are used within `React App` won't be wrapped with this nor with the store